src/main/java/com/wisedevlife/whytalkuser/service/implement/MatchingServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+package com.wisedevlife.whytalkuser.service.implement;
+
+import com.wisedevlife.whytalkuser.common.constants.MqConstant;
+import com.wisedevlife.whytalkuser.common.constants.RedisConstant;
+import com.wisedevlife.whytalkuser.common.utils.MessageQueueUtils;
+import com.wisedevlife.whytalkuser.common.utils.RedisUtils;
+import com.wisedevlife.whytalkuser.common.utils.UserProfileUtil;
+import com.wisedevlife.whytalkuser.dto.message.MatchingJobMessage;
+import com.wisedevlife.whytalkuser.dto.request.MatchingUserMeta;
+import com.wisedevlife.whytalkuser.entity.UserProfile;
+import com.wisedevlife.whytalkuser.model.MatchingFilter;
+import com.wisedevlife.whytalkuser.service.MatchingService;
+import com.wisedevlife.whytalkuser.service.UserProfileService;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Primary
+@Service
+@RequiredArgsConstructor
+public class MatchingServiceImpl implements MatchingService {
+    private final RedisUtils redisUtils;
+    private final MessageQueueUtils messageQueueUtils;
+
+    private final UserProfileService userProfileService;
+    private final UserProfileUtil userProfileUtil;
+
+    @Override
+    public void startMatchingProcess(String userId, MatchingFilter filter) {
+        Instant startMatchingTime = Instant.now();
+        long startMatchingTimestamp = startMatchingTime.getEpochSecond();
+
+        UserProfile user = userProfileService.getUserProfile(userId);
+
+        MatchingUserMeta matchingUserMeta =
+                MatchingUserMeta.builder()
+                        .userProfile(user)
+                        .matchingFilter(filter)
+                        .startMatchingTimestamp(startMatchingTimestamp)
+                        .build();
+
+        redisUtils.setObject(RedisConstant.getUserProfileKey(userId), matchingUserMeta);
+
+        redisUtils.addValueToSortedSet(
+                RedisConstant.MATCHING_POOL_SET, userId, (double) startMatchingTimestamp);
+        log.info(
+                "put user {} into matching pool with score {}",
+                userId,
+                (double) startMatchingTimestamp);
+
+        MatchingJobMessage matchingJobMessage = MatchingJobMessage.builder().userId(userId).build();
+
+        messageQueueUtils.publishMessageToQueue(MqConstant.MATCHING_QUEUE, matchingJobMessage);
+        log.info("enqueue user {}", userId);
+    }
+
+    @Override
+    public List<UserProfile> matchUser(String currentUserId) {
+        if (!redisUtils.isSortedSetContainsKey(RedisConstant.MATCHING_POOL_SET, currentUserId)) {
+            throw new RuntimeException(
+                    String.format("User %s is not in matching status", currentUserId));
```

Comment:
What do you think about to introduce a `MatchException`, which could have a common prefix for logging user's information, and a `MatchErrorCode` enum to get the common error code and message?